runtime/tests/tests.rs

```diff
@@ -419,23 +325,27 @@ fn staking_reward_should_work() {
 		.validator_count(validators.len())
 		.build()
 		.execute_with(|| {
-			// There are 12 accounts: 6 pre-configured + 6 stash
-			assert_eq!(GenericAsset::total_issuance(CENNZ_ASSET_ID), balance_amount * 12);
-			assert_eq!(GenericAsset::total_issuance(CENTRAPAY_ASSET_ID), balance_amount * 12);
+			assert_eq!(GenericAsset::total_issuance(CENNZ_ASSET_ID), total_issuance);
+			assert_eq!(GenericAsset::total_issuance(CENTRAPAY_ASSET_ID), total_issuance);
 
 			start_era(1);
-			// TODO: assert total issuance
 			let validator_len = validators.len() as Balance;
-			let total_reward = reward_validators(&validators);
+			let (total_reward, inflation) = reward_validators(&validators);
 			let per_staking_reward = total_reward / validator_len;
 			// The balance of stash accounts remain the same within the same era
 			for validator in validators.clone() {
 				let (stash, _) = validator;
 				check_free_balance_updated(&stash, balance_amount);
 			}
 
-			start_era(2);
-			// TODO: assert total issuance
+			// Advance a session to begin era 2
+			advance_session();
+			assert_eq!(GenericAsset::total_issuance(CENNZ_ASSET_ID), total_issuance);
+			assert_eq!(
+				GenericAsset::total_issuance(CENTRAPAY_ASSET_ID), // FIXME: 33_000_000 is coming from a weird timing issue between sessions
+				total_issuance + inflation + 33_000_000, // FIXME: Changing era_duration in fn current_total_payout outputs different inflation amount (but not sure what it should be)
```

Comment:
Tried to figure out why we end up issuing `33_000_000` much more cennz (which is more than the max_payout), but still not sure. It's definitely something to do with the `era_duration` in `fn current_total_payout` (changing this results in different inflation amount).

Reply:
If you refer to the fn new_era(), the total_reward (first parameter returned from compute_total_payout()) is what is actually used to reward the validators. The max is only used to ensure the total_payout doesn't go over the max. 
therefore I think the test here should use total_reward to calculate total issuance 
new_issuance = total_reward / validator_len * validator_len; //this also takes care of remainders

Reply:
what's leftover from given out is actually burned
```let rest = _max_payout.saturating_sub(total_payout);
Self::deposit_event(RawEvent::Reward(total_payout, rest));```

Reply:
What's leftover is *not* burned. It's issued using T::RewardCurrency (cpay) back to the T::RewardRemainder (which is currently configured as treasury):
https://github.com/cennznet/cennznet/blob/19ca3f16c6ceb9f56df388d320bfc002a32c50cc/crml/staking/src/lib.rs#L1439
(also see this commit that shows the related system events: https://github.com/cennznet/cennznet/pull/153/commits/21791f30e81573fb46af086e7d6bd8e272ac6f37)

------

https://github.com/cennznet/cennznet/blob/19ca3f16c6ceb9f56df388d320bfc002a32c50cc/crml/staking/src/lib.rs#L1413
^ The variable names are confusing but `max_payout` not only ensures that you don't pay over that amount, but also it's the inflation target that we are trying to achieve. Hence `total_payout` is paid out to the validators and nominators and then the `rest` is calculated and issued to the `T::RewardRemainder`:
https://github.com/cennznet/cennznet/blob/develop/crml/staking/src/lib.rs#L1435-L1439

Reply:
And if it helps the numbers are:
```rust
let (total_reward, inflation) = reward_validators(&validators);
assert_eq!(total_reward, 711_000_000);
assert_eq!(inflation,  1_896_000_000);
```

Reply:
I'm gonna look into this a bit deeper.
@cowboy-bebug makes some interesting points - it might be that the currency overall should have a fixed inflation, but we will reward less than that

Reply:
Uhh yes you are right! the unbalance is implemented for the treasury:
impl<T: Trait> OnUnbalanced<NegativeImbalanceOf<T>> for Module<T>

and it puts the money into the treasury account:
let _ = T::Currency::resolve_creating(&Self::account_id(), amount);

I'm surprised at the difference between the two numbers though - like you said, it may have something to do with the duration of the era.

Reply:
I got a nice tip from @hoani to try out and resolve this issue. I'll mark it as resolved for now 😄

Reply:
As discussed in team planning, fixing this is split into another task, and will be done in another PR.